### PR TITLE
[FIX][mail_environment][server_environment] Prevent ImportError.

### LINKED ISSUE #488: Depends server_environment missed
The module server_environment depends of server_environment_files, it not exist. ¿Maybe server_environment_files_samples?

### PR DESCRIPTION
**Pull request by yajo**
Fixes #488.

It should fix Travis in OCA/website, like in https://github.com/OCA/website/pull/303#issuecomment-268797438.

@Tecnativa

### CODE REVIEW

**Review by pedrobaeza — APPROVED** [review]

File: mail_environment/__init__.py [hunk 1]
```diff
@@ -1,2 +1,10 @@
 # -*- coding: utf-8 -*-
-from . import env_mail
+import logging
+
+_logger = logging.getLogger(__name__)
+
+try:
+    from . import env_mail
+except ImportError:
+    _logger.info("ImportError raised while loading module.")
```
Comment (petrus-v) [inline mail_environment/__init__.py:9]: Could be nice to tell which module is not able to be imported!

suggested:

```
   _logger.info("ImportError raised while loading env_mail module.")

File: server_environment/__init__.py [hunk 2]
```diff
@@ -18,4 +18,12 @@
...
 ##############################################################################
-from .serv_config import serv_config, setboolean
+import logging
+
+_logger = logging.getLogger(__name__)
+
+try:
+    from .serv_config import serv_config, setboolean
+except ImportError:
+    _logger.info("ImportError raised while loading module.")
```
Comment (petrus-v) [inline server_environment/__init__.py:28]: same comment here!

**Review by petrus-v — APPROVED** [review]
small typo otherwise you get my :+1:

File: mail_environment/__init__.py (hunk 1 above)
Comment (yajo) [reply mail_environment/__init__.py:9]: No need, the logs already say where do error come from, that's implicit by `getLogger(__name__)`

**Comment by petrus-v** [comment]
In fact the real matter is the missing dependency(sever_environement_files) for server_environment, isn't it?

**Comment by yajo** [comment]
But AFAICS that addon [should be created manually](https://github.com/OCA/server-tools/tree/9.0/server_environment#installation) in each environment.

OTOH, the real problem is [not controlling the ImportError](https://github.com/OCA/pylint-odoo/pull/68). That's a bug.

**Comment by petrus-v** [comment]
ok thanks for those informations! i've never used this module :) I was reading that on the history on other PR...

**Comment by yajo** [comment]
Me neither, but this bug is messing up other repos' bots... 😆

**Review by dreispt — APPROVED** [review]

**Comment by dreispt** [comment]
@Yajo sorry for nagging, but you shouldn't merge your own PRs.

**Comment by DarioLodeiros** [comment]
the hard import: https://github.com/OCA/server-tools/blob/8.0/server_environment/serv_config.py#L32
is the problem, when you run Odoo, this import is a real problem becouse it causes boot failure even if it is not installed.
I think that isn't enought this fix... ¿?

**Comment by moylop260** [comment]
What about use the new standard of ImportError to know the exactly package with the error?

https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#importerror
```bash
try:
    import external_dependency_python_N
    import external_dependency_python_M
    EXTERNAL_DEPENDENCY_BINARY_N_PATH = tools.find_in_path('external_dependency_binary_N')
    EXTERNAL_DEPENDENCY_BINARY_M_PATH = tools.find_in_path('external_dependency_binary_M')
except (ImportError, IOError) as err:
    _logger.debug(err)
```

**Comment by yajo** [comment]
Hi @dreispt, this had already 3x👍; isn't that enough to merge? (Sorry if I missed something, I've been PSC for not so long).

@DarioLodeiros, I tried that. The problem is that if you catch the ImportError but still continue executing the whole module, another exception would be probably raised because the module was not loaded properly. I decided to catch the exception at a higher level to abort loading the rest of the file(s) if the import fails. After all, this is a quick patch for an addon that I don't use but breaks my environment, its actual developers should do a better patch if they need it 😉.

@moylop260 If I understand fine, you mean that you want to get the ImportError traceback. Actually I'm using a feature of the logging module, [the `exc_info=True` flag](https://github.com/OCA/server-tools/pull/666/files#diff-81332b35f063c24510597cd358dfcd03R10), which outputs current exception traceback automatically appended to the message. I use an INFO to display the import error, but leave the traceback at DEBUG level. Also, [all imports with possible failures are grouped](https://github.com/OCA/server-tools/pull/666/files#diff-e50635d23d671f97c9c8592d7f053a1dR26), so what exactly are you missing here? 😕

**Comment by sbidoul** [comment]
@Yajo why was this module suddenly creating problems? It had not been causing any problem for years.

**Comment by yajo** [comment]
No idea, but [it happened](https://github.com/OCA/website/pull/303#issuecomment-269050419).

**Comment by DarioLodeiros** [comment]
The source of the problem is the hard import of a module that does not exist... server_environment_files
If you clone the repository, actually it break the enviroment... i think that thisn't admissible... ¿?¿

**Comment by yajo** [comment]
Indeed. I mean I don't know how come it was not failing before 😁

**Comment by sbidoul** [comment]
`mail_environment` was never auto imported (unless installed in a database) because it has no `static` directory. So there is no reason it would start failing suddenty. @Yajo I could not find the travis failure you mention. 

#617 created some noise because that module had a `static` directory causing it to be auto imported. But that is fixed in #676. So please guys do not rush at the risk of doing more harm than good.

**Comment by yajo** [comment]
Nice to know that, although could you please detail what did I harm here?

**Pull request merged**